src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,115 @@
+    ) {
+        User currentUser = getCurrentUser().getUser();
+        Iterable<RecommendationRequest> recommendationRequests = recommendationRequestRepository.findAllByProfessorId(currentUser.getId());
+        return recommendationRequests;
+    }
+
+    /**
+     * This method returns a single recommendation request where the current user is either the requester or the professor.
+     * @param id id of the Recommendation Requests to get
+     * @return a single recommendation request where the current user is either the requester or the professor
+     */
+    @Operation(summary = "Get a single recommendation request where the current user is either the requester or the professor")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public RecommendationRequest getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+            Long currentUserId = getCurrentUser().getUser().getId();
+            RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+            if (recommendationRequest.getRequester().getId() != currentUserId && recommendationRequest.getProfessor().getId() != currentUserId) {
+                throw new EntityNotFoundException(RecommendationRequest.class, id);
+            }
+        return recommendationRequest;
+    }
+
+    /**
+     * This method creates a new Recommendation Request. Accessible only to users with the role "ROLE_USER" so professors and students can both create.
+     * @param professorId id from a dropdown of professors from the form in create page
+     * @param recommendationType recommendation types of request
+     * @param details details of request
+     * @param dueDate submission date of request
+     * @return the save recommendationrequests (with it's id field set by the database)
+     */
+    @Operation(summary = "Create a new recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequests(
+            @Parameter(name = "professorId") @RequestParam Long professorId,
+            @Parameter(name = "recommendationType") @RequestParam String recommendationType,
+            @Parameter(name = "details") @RequestParam String details,
+            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime dueDate)
+            {
+        //get current date right now and set status to pending
+        CurrentUser currentUser = getCurrentUser();
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+        User professor = userRepository.findById(professorId).orElseThrow(() -> new EntityNotFoundException(User.class, professorId));
+        recommendationRequest.setProfessor(professor);
+        // recommendationRequest.setProfessor_id(professorId);
```

Comment:
```suggestion

```

Remove commented out code please.
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed